app/src/adapters/db/engine/db_engine.py

```diff
@@ -0,0 +1,18 @@
+import abc
+
+import sqlalchemy
+
+
+class DbEngine(abc.ABC, metaclass=abc.ABCMeta):
+    @abc.abstractmethod
+    def build_engine(self) -> sqlalchemy.engine.Engine:
+        pass
+
+    @abc.abstractmethod
+    def check_db_connection(self, conn: sqlalchemy.engine.Connection) -> None:
```

Comment:
nit, we currently define the type `Connection = sqlachemy.engine.Connection` in `client.py`, but now that we're extracting out the engine.py module, the Connection concept belongs more with the engine.py module rather than in the client.py module, so let's move that type definition to this file and simplify this type hint as

```suggestion
    def check_db_connection(self, conn: Connection) -> None:
```